### DISCUSSION TITLE
Indicator HA update

### DISCUSSION THREAD

**Discussion by Love4yzp** [Seeed-Solution/SenseCAP_Indicator_ESP32#29] [Q&A]
### Added

- implement console feature for indicator_ha example to set broker url and username and password.

<hr /><em>This discussion was created from the release <a href='https://github.com/Seeed-Solution/SenseCAP_Indicator_ESP32/releases/tag/v1.0.2'>Indicator HA update</a>.</em>

**Comment by Love4yzp** [comment]
### Instructions for Flashing the BIN File

If you're utilizing esptool to facilitate the firmware flashing process, refer to the [flashing-firmware documentation](https://wiki.seeedstudio.com/SenseCAP_Indicator_How_To_Flash_The_Default_Firmware/#flashing-firmware) for comprehensive details.

Alternatively, you have two options for obtaining the necessary tool:

1. Download it from the [official repository](https://github.com/espressif/esptool).
2. Utilize the tool I've provided, available [here](https://github.com/Love4yzp/ESP32-Tools). This tool eliminates the need for an ESP-IDF environment for compilation and flashing.

The BIN file release has been pre-processed using `build.bat`, making it straightforward to flash using `flash.bat` for Windows users. Remember to rename the file to `firmware.bin`, as line 42 references a file named `firmware.bin` for flashing.

Mac users, on the other hand, should use the following command:

```bash
./esptool --chip esp32s3 --baud 144000 write_flash -z 0x0 firmware.bin
```

> You can adjust the baud rate according to your desired flash speed.

**Comment by Love4yzp** [comment]
# Mannu CLI Commands

View the implementation in `indicator_cmd.c` .

## `haconfig`

Displays the current MQTT configuration.

## `setmqtt`

Configure MQTT settings:

- `-u, --usr=<username>`: Set MQTT username for authentication.
- `-p, --psw=<password>`: Specify MQTT password for security.
- `-a, --addr=<broker_url>`: Define MQTT broker URL for messaging.
- `-c, --id=<client_id>`: Configure MQTT client ID for device identification.

Example:

```shell
setmqtt -u indicator_username
setmqtt -p indicator_psw
setmqtt -a test.mosquitto.org
setmqtt --addr broker.emqx.io
setmqtt -u name -p 1234
```

## `help`

Access a list of available commands with `help`.

**Comment by MarcLuethi8356** [comment]
Hi

I tried your firmware for the SenseCAP. 
I could connect it to the home assistant. Great thanks.

The only problem I have, is that I have more than 1 such device. 
This means that I will have the same topic "indicator/sensor" for each device. 
Would it be possible for you to update your software, so I can also change the "indicator" topic to some other value using the terminal?
So the topics would be something like "sensecapoffice1" or "sensecapoffice2".

For example 
setmqtt -t sensecapoffice1

Marc
